File: hail/python/hail/ir/export_type.py
from hail.typecheck import enumeration


class ExportType:
    CONCATENATED = "concatenated"
    PARALLEL_SEPARATE_HEADER = "separate_header"
    PARALLEL_HEADER_IN_SHARD = "header_per_shard"
    PARALLEL_COMPOSABLE = "composable"

    checker = enumeration("concatenated", "separate_header", "header_per_shard",
                          "composable")

    @staticmethod
    def default(export_type):
        if export_type is None:
            return ExportType.CONCATENATED
        else:
            return export_type

File: hail/hail/src/is/hail/expr/ir/ExportType.scala
package is.hail.expr.ir

object ExportType {
  val CONCATENATED = "concatenated"
  val PARALLEL_SEPARATE_HEADER = "separate_header"
  val PARALLEL_HEADER_IN_SHARD = "header_per_shard"
  val PARALLEL_COMPOSABLE = "composable"
}
